task17/index.js

```diff
@@ -0,0 +1,127 @@
+const URL = "https://mocki.io/v1/af7d1b29-da7d-4eba-977b-cdb5caeb2fff";
+
+const slideShowContainer = document.getElementById("js-slideShowContainer");
+const slideShowWrapper = document.getElementById("js-slideShowWrapper");
+const slideImgList = document.getElementById("js-slideImgList");
+
+const createElementWithClassName = (element, name) => {
+  const createdElement = document.createElement(element);
+  createdElement.classList.add(name);
+  return createdElement;
+};
+
+const addLoading = () => {
+  const loadingPlace = createElementWithClassName("div", "loader");
+  loadingPlace.id = "js-loadingPlace";
+  slideShowWrapper.appendChild(loadingPlace);
+};
+
+const removeLoading = () => document.getElementById("js-loadingPlace").remove();
+
+const renderErrorMessage = (errorMessage) => {
+  slideShowWrapper.textContent = errorMessage;
+};
+
+const getFetchData = async () => {
+  const response = await fetch(URL);
+  if (!response.ok) {
+    console.error(`${response.status}:${response.statusText}`);
+    renderErrorMessage("問題が発生し表示することができません。");
+  }
+  const json = await response.json();
+  return json;
+};
+
+const fetchImgData = () => {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve(getFetchData(URL));
+    }, 3000);
+  });
+};
+
+const init = async () => {
+  addLoading();
+  let slideImgData;
+  try {
+    slideImgData = await fetchImgData();
+  } catch (error) {
+    console.error(error.message);
+    renderErrorMessage("表示することができません");
+  } finally {
+    removeLoading();
+  }
+  currentImgIndex = slideImgData.findIndex((v) => v.display);
+  renderSlideItem(slideImgData);
+  renderSlideNumber(slideImgData);
+  addEventListenerForBtn();
+  toggleDisabledOfButton(slideImgData);
+};
+init();
+
+const renderSlideItem = (slideImgData) => {
+  const slideImgFragment = document.createDocumentFragment();
+  for (let i = 0; i < slideImgData.length; i++) {
+    const slideImgItem = createElementWithClassName("li", "slide-img-item");
+    const slideImage = createElementWithClassName("img", "slide-image");
+    slideImage.src = slideImgData[i].img;
+    slideImage.alt = slideImgData[i].alt;
+    slideImgData[i].display && slideImgItem.classList.add("is-active");
+    slideImgFragment.appendChild(slideImgItem).appendChild(slideImage);
+  }
+  slideImgList.appendChild(slideImgFragment);
+  createSlideBtn();
+};
+
+const createSlideBtn = () => {
+  const slideShowItem = document.getElementById("js-slideShowItem");
```

Comment:
IMO
Since there is no return value, `render**` seems more appropriate.
```suggestion
const renderSlideBtn = () => {
  const slideShowItem = document.getElementById("js-slideShowItem");
```

Reply:
Thank you for your time, I didn't quite understand how to write the arguments. I will study this area more.
c876b8c